### DISCUSSION TITLE
Windows Sandboxing

### DISCUSSION THREAD

**Discussion by xyzeva** [Minecraft-Malware-Prevention-Alliance/working-groups#10] [Sandboxing]
# Issue

Windows offers barely any sandboxing kernel api calls, etc. This prevents us from sandboxing windows kernel-level and have to go for other solutions like MSIX

### Current Options

- [AppContainer's](https://learn.microsoft.com/en-us/windows/win32/secauthz/appcontainer-isolation)
- [Chromium-like approach](https://chromium.googlesource.com/chromium/src/+/HEAD/docs/design/sandbox.md#Sandbox-Windows-architecture)

**Comment by xyzeva** [comment]
Can be bypassed very easily

**Comment by getchoo** [comment]
msix packaging (without sandboxing yet) is being worked on in https://github.com/PrismLauncher/PrismLauncher/pull/1149. this includes some more advanced things like integrating msixs for multiple architectures in an msixbundle, as well as allowing for auto updates with appinstaller. i hope this can be a good example for some other projects to get started :)

[this](https://github.com/microsoft/win32-app-isolation) work from microsoft also seems very interesting...but sadly isn't even in regular windows 11 releases yet, let alone any older versions :/

**Comment by TreeDB** [comment]
There is Sandboxie (was closed sourced for a while, got bought out, then now is open source) which sandboxes applications: https://github.com/sandboxie-plus/Sandboxie

It has been a while since I have used this software though, but may point us in the right direction

**Reply by chexo3** [reply]
Relevant pages of Sandboxie’s documentation:

https://sandboxie-plus.github.io/sandboxie-docs/Content/IsolationMechanism.html

https://sandboxie-plus.github.io/sandboxie-docs/Content/TokenMagic.html

These discuss the isolation mechanism.
